python/pyspark/rdd.py

```diff
@@ -84,7 +84,7 @@ def portable_hash(x):
         h ^= len(x)
         if h == -1:
             h = -2
-        return h
+        return int(h)
```

Comment:
The `h` become a long when `h *= 1000003`, even after `h &= sys.maxsize` or (maxint). 

The fix looks good to me.